plugins/org.eclipse.sirius.tests.swtbot/src/org/eclipse/sirius/tests/swtbot/GroupElementsInOneOtherTests.java

```diff
@@ -292,7 +294,23 @@ protected void tearDown() throws Exception {
     private void checkLocationAlignOnGrid(Point location, String elementNameToDisplay, int gridSpacing) {
         boolean locationIsOK = (location.x % gridSpacing) == 0 || (location.y % gridSpacing) == 0;
         if (!locationIsOK) {
-            fail("For " + elementNameToDisplay + ", the x or y coordinate of the top left corner should be on the grid (grid spacing = " + gridSpacing + "), but was: " + location + ".");
+            fail("For " + elementNameToDisplay + ", the x or y coordinate of the top left corner should be on the grid (grid spacing = " + gridSpacing + "), but was: " + location + "."); //$NON-NLS-1$ //$NON-NLS-2$ //$NON-NLS-3$ //$NON-NLS-4$
+        }
+    }
+
+    private Point getEditorCenter() {
+        PrecisionPoint editorCenter = new PrecisionPoint(getEditorVisiblePart(diagramEditPartBot.part()).getCenter());
+        Object helper = diagramEditPartBot.part().getAdapter(SnapToHelper.class);
+        if (helper instanceof SnapToHelper snapToHelper) {
+            PrecisionPoint preciseLocation = new PrecisionPoint(getEditorVisiblePart(diagramEditPartBot.part()).getCenter());
+            snapToHelper.snapPoint(new CreateRequest(), PositionConstants.HORIZONTAL | PositionConstants.VERTICAL, preciseLocation, editorCenter);
         }
+        return editorCenter.getCopy();
+    }
+
+    private Point getAvailablePosition(Rectangle rectangle) {
```

Comment:
I don't know if we could find a better way to achieve that but the unit test is using the API that we want to test to verify the result?

Reply:
Done